applications/FROST.md

```diff
@@ -0,0 +1,265 @@
+
+The current multisig implementation with a threshold $t$ does not scale well with the number of participants because the protocol is done on-chain: a participant that wants to approve a given call needs to sign that call and send it to the network to be validated, having to pay fees for that. When there are sufficient call approvals ($t - 1$), a final call is made by the last participant representing the multisig account. Therefore, you always need to send at least $t$ extrinsics to the network, which is costly. Besides that, the network knows exactly which accounts approved the call, which is not good for privacy.
+
+Our proposal solves those issues because the protocol is done off-chain instead of on-chain. The protocol starts with a Distributed Key Generation, where a secret key is generated that no single participant knows of. Instead, the participants generate shares of that secret key, that they can later use to sign an extrinsic. This is done by using the Pedersen's variation of the Shamir's secret sharing scheme, which allows the participants to verify their shares and does not require a trusted dealer. Our solution is:
+- **More scalable and cheaper:** Only a single Schnorr signature is submitted to the network and is independent of the number of signers.
+- **More user-friendly:** A participant does not necessarily need to make the existential deposit on its account nor pay to send an extrisinc (the broadcaster account needs to pay for it, but how and if the fee is divided among participants will depend on the use case).
+- **More private:** The output Schnorr signature is indistinguishable from any other signature on the network, and the signers are not known because the signing protocol happens off-chain.
+
+### Arquitecture
+
+This section describes in a simplified way how the FROST protocol works and how it will be implemented to be compatible with Substrate. For details about the protocol, consult the original paper (https://eprint.iacr.org/2020/852) and its Rust implementation (https://github.com/ZcashFoundation/frost).
+
+The FROST protocol is composed of two subprotocols:
+- The Distributed Key Generation protocol, which generates a pair of keys (secret and public) shared between the participants in the protocol. No entity knows the secret key.
+- The Threshold Signing protocol, which generates a Schnorr signature corresponding to the public key generated in the DKG protocol from at least $t$ partial signatures of the participants.
+
+The high level flow of the DKG protocol is the following:
+1. One participant sends the message to start the DKG to the server, which includes the addresses of the participants, the threshold $t$ and a commitment random polynomial of degree $t - 1$.
+2. The server verifies the message, asks if the other wallets want to participate in the protocol and, if yes, sends the first message to them.
```

Comment:
This would happen off-chain, yes. The network protocol used would be Secure WebSockets (WSS). The idea is that each client (wallet), if interested, would connect to the server (whose address would be hardcoded in the wallet) through an authenticated channel and the server would do the mapping between the public key addresses (accounts) and the network addresses of the clients, so that it can forward the messages between them.